src/librustc/diagnostics.rs

```diff
@@ -1941,36 +1950,34 @@ or isn't assigned to a constant expression. Example:
 static foo: i32 = 42;
 static bar: i32 = 43;
 
-let baz: bool = { (&foo as *const i32) == (&bar as *const i32) };
+let baz: bool = { (&foo as *const i32) < (&bar as *const i32) };
 // baz isn't a constant expression so it's ok
 ```
 "##,
 
 E0396: r##"
-The value assigned to a constant expression must be known at compile time,
-which is not the case when dereferencing raw pointers. Erroneous code
-example:
+The value behind a raw pointer can't be determined at compile-time
+(or even link-time), which means it can't be used in a constant
+expression.
 
+For example:
 ```
-const foo: i32 = 42;
-const baz: *const i32 = (&foo as *const i32);
+const REG_ADDR: *const u8 = 0x5f3759df as *const u8;
 
-const deref: i32 = *baz;
+const VALUE: u8 = unsafe { *REG_ADDR };
 // error: raw pointers cannot be dereferenced in constants
 ```
 
-To fix this error, please do not assign this value to a constant expression.
-Example:
+If you wanted to determine the value of the raw pointer at run-time,
+please evaluate it at run-time instead.
```

Comment:
"If you want it at run-time, do it at run-time". This way to say it seems a bit weird.